src/manager.cpp

```diff
@@ -127,12 +131,14 @@ void Manager::stop(bool discard)
     }
 }
 
-void Manager::dump(const Log& log, const std::string& channel_name, Logger::ContextInfo const& context_info)
+void Manager::dump(const Log& log, const std::string& channel_name, ContextInfo const& context_info)
 {
+    // Copy shared pointer in order to allow update without locking on set_global_context_info
+    auto context_info_handle(global_context_info_);
```

Comment:
can we maybe just use weak_ptr?

Reply:
no, we want this as owner of the pointer.
think of the following:
thread 1: writing log
thread 1: copy shared ptr
thread 2: setting new context == updating the shared ptr count
thread 1: accessing old context.

If this was not a shared pointer on this point it could be deleted